### DISCUSSION TITLE
Matrix inequality on a 3D sdpvar

### DISCUSSION THREAD

**Discussion by mkdeak** [yalmip/YALMIP#1420] [General discussion] [answered]
Hi- 

I understand that YALMIP detects symmetry in order to decide whether to treat the constraint `A >= 0` as an element-wise inequality or a matrix inequality. 

Suppose I declare a 3 dimensional sdpvar which consists of k pages of `n by n` matrices:

```
As = sdpvar(n, n, k);
```

Now every page `As(:,:,i)` of `As` is an symmetric matrix variable, however the constraint `As >= 0` gets interpreted as an element-wise inequality. How can I have this constraint interpreted as a semidefinite constraint on each of the pages of `As`? 

A second question I have is this: assume that `B` is some fixed `n by n` matrix which is symmetric. I would like to further assert that `B * As(:, :, i) >= 0` for each index `i`. I have already implemented a helper function that does the matrix multiplication page-wise by reshaping `As`. So suppose `my_pagemtimes` is this helper function. I would like the assertion

```
my_pagemtimes(B, As) >= 0
```

to be regarded as another matrix inequality instead of element-wise inequalities. Since `B` is symmetric and so are all the pages of `As`, each page of the left hand side is symmetric here as well. Yet YALMIP regards this as an element-wise inequality. How can I make this work?

**Comment by johanlofberg** [comment, marked as answer]
That is probably not intended behavior, and will have to be fixed. For now, you will have to loop over the slices

B*A where B and A are symmetric does not mean B*A is symmetric. Here you will have to loop regardless though, as nd-array matrix multiplications simply isn't supported

**Reply by mkdeak** [reply]
Thanks for the quick reply. And yes, I got that wrong for the second question... thanks for pointing it out.
